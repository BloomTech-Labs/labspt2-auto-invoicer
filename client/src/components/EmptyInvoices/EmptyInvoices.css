.emptyInvoices {
  width: 420px;
  height: 400px;
  margin-top: -40px;
}
.topText {
  margin-top: 15px;
  font-size: 1.8rem;
margin-left:20px;

}
.btmText {
  font-size: 1.8rem;
  margin-top: -20px;
  margin-left:40px;
}
.align {
  display: flex;
  justify-content: center;
}
@media (max-width: 500px) {
  .emptyInvoices {
    width: 95%;
  }
  .topText {
    margin-left: 20px;
  }
  .btmText {
    margin-left: 40px;
  }
}
