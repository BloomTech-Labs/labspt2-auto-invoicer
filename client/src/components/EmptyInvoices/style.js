export default theme => ({
  root: {
    width: "95%",
    margin: "20px auto",
    height:"1500",
    marginTop: theme.spacing.unit * 3,
    [theme.breakpoints.up("sm")]: {
      width: "100%",
      margin: "0px",
      marginTop: theme.spacing.unit * 3
    },
  },
  button: {
    color: "#ffffff",
    backgroundColor: "#8bc34a",
    "&:hover": {
      backgroundColor: "#2d2f31"
    },
  },
  toolbar: {
    display: "flex",
    justifyContent: "space-between"
  },
  table: {
    minWidth: 1000,
    [theme.breakpoints.up("sm")]: {
      width: "100%",
      minWidth: 0
    }
  },
  tableWrapper: {
    overflowX: "auto"
  },
  grow: {
    flexGrow: 1
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  },
  title: {
    fontSize: 30,
    [theme.breakpoints.up("sm")]: {
      display: "block",
      fontSize: 45
    }
  }
});
