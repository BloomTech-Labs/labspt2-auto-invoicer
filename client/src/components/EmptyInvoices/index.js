import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";
import Grow from "@material-ui/core/Grow";
import AppBar from "@material-ui/core/AppBar";
import Button from "@material-ui/core/Button";
import Toolbar from "@material-ui/core/Toolbar";
import { ReactComponent as Investing } from "../../assets/undraw_investing_7u74.svg";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import styles from "./style";

// import css here
import "./EmptyInvoices.css";

const EmptyInvoices = props => {
  const [buttonSize, setButtonSize] = useState("large");
  const buttonSizeFunc = () => {
    window.innerWidth > 500 ? setButtonSize("large") : setButtonSize("medium");
  };
  useEffect(() => {
    buttonSizeFunc();
  }, []);

  const { classes, userID } = props;
  return (
    <Grow in={true} {...{ timeout: 1300 }}>
      <Paper
        style={{
          height: "490px",
          marginBottom: "20px"
        }}
        className={classes.root}
      >
        <AppBar
          style={{ backgroundColor: "#ffffff", color: "#2d2f31" }}
          position="static"
        >
          <Toolbar className={classes.toolbar}>
            <Typography
              //className={classes.title}
              variant="h3"
              color="inherit"
              noWrap
            >
              Invoices
            </Typography>
            <Link to={`/user/${userID}/invoice/create`}>
              <Button
                variant="contained"
                className={classes.button}
                size={buttonSize}
              >
                Create
              </Button>
            </Link>
          </Toolbar>
        </AppBar>
        <div className="align">
          <Link to={`/user/${userID}/invoice/create`}>
            <p className="topText">
              Your Path To Financial Success Starts Here.
            </p>
            <Investing className="emptyInvoices" />
            <p className="btmText">Click Now To Create Your First Invoice.</p>
          </Link>
        </div>
      </Paper>
    </Grow>
  );
};

export default withStyles(styles)(EmptyInvoices);
