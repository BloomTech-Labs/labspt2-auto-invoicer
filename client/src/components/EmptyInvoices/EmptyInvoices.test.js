import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import EmptyInvoices from "../EmptyInvoices";

describe("<EmptyInvoices />", () => {
  it("renders Empty Invoices", () => {
    const { asFragment } = render(<EmptyInvoices />);
    expect(asFragment()).toMatchSnapshot();
  });
});
