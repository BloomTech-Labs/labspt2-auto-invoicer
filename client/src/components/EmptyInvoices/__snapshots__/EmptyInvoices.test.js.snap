// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyInvoices /> renders Empty Invoices 1`] = `
<DocumentFragment>
  <section
    class="invoices-container empty-invoices"
  >
    <article
      class="invoice-card add-invoice"
    >
      <h3>
        Add a New Invoice
      </h3>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
    </article>
  </section>
</DocumentFragment>
`;
